dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java

```diff
@@ -832,27 +834,59 @@ public ImportSummaries deleteEvents( List<String> uids, boolean clearSession )
     // HELPERS
     // -------------------------------------------------------------------------
 
-    private List<OrganisationUnit> getOrganisationUnits( EventSearchParams params )
+    private List<OrganisationUnit> getProvidedOrUserAssignedOrganisationUnits( EventSearchParams params,
+        User currentUser )
     {
         List<OrganisationUnit> organisationUnits = new ArrayList<>();
 
         OrganisationUnit orgUnit = params.getOrgUnit();
         OrganisationUnitSelectionMode orgUnitSelectionMode = params.getOrgUnitSelectionMode();
 
-        if ( params.getOrgUnit() != null )
+        if ( currentUser.isSuper() )
         {
+            if ( params.getOrgUnit() != null )
+            {
+                if ( OrganisationUnitSelectionMode.DESCENDANTS.equals( orgUnitSelectionMode ) )
+                {
+                    organisationUnits
+                        .addAll( organisationUnitService.getOrganisationUnitWithChildren( orgUnit.getUid() ) );
+                }
+                else if ( OrganisationUnitSelectionMode.CHILDREN.equals( orgUnitSelectionMode ) )
+                {
+                    organisationUnits.add( orgUnit );
+                    organisationUnits.addAll( orgUnit.getChildren() );
+                }
+                else // SELECTED
+                {
+                    organisationUnits.add( orgUnit );
+                }
+            }
+        }
+        else
+        {
+            Set<OrganisationUnit> userOrgUnits = params.getOrgUnit() == null
+                ? currentUser.getDataViewOrganisationUnitsWithFallback()
+                : Sets.newHashSet( params.getOrgUnit() );
+
+            List<String> orgUnitIds = userOrgUnits.stream()
+                .map( BaseIdentifiableObject::getUid ).collect( Collectors.toList() );
+
             if ( OrganisationUnitSelectionMode.DESCENDANTS.equals( orgUnitSelectionMode ) )
             {
-                organisationUnits.addAll( organisationUnitService.getOrganisationUnitWithChildren( orgUnit.getUid() ) );
+                organisationUnits.addAll( organisationUnitService.getOrganisationUnitsWithChildren( orgUnitIds ) );
             }
             else if ( OrganisationUnitSelectionMode.CHILDREN.equals( orgUnitSelectionMode ) )
             {
-                organisationUnits.add( orgUnit );
-                organisationUnits.addAll( orgUnit.getChildren() );
+                Set<OrganisationUnit> children = userOrgUnits.stream()
+                    .map( OrganisationUnit::getChildren )
+                    .flatMap( Collection::stream )
+                    .collect( Collectors.toSet() );
+
+                organisationUnits.addAll( children );
             }
-            else // SELECTED
+            else
```

Comment:
It seems we need to implement different behavior for OrganisationUnitSelectionMode ACCESSIBLE, CAPTURE and ALL?

Reply:
https://github.com/dhis2/dhis2-core/pull/8711